common/models/resource_provider.go

```diff
@@ -0,0 +1,8 @@
+package models
+
+type ResourceProvider struct {
+	ID              string `json:"id"`
+	InUse           bool   `json:"in_use"`
+	UsedPorts       []int  `json:"used_ports"`
```

Comment:
just noticed this is a slice of `int`s... we are not tracking protocol type here, are we? Unless I'm missing something, this would disallow the same port if used by UDP and TCP, which is the case for Consul for example (8500 tcp and udp are needed)